examples/hackernews/skip_leader/src/hackernews.service.ts

```diff
@@ -0,0 +1,130 @@
+import type {
+  Context,
+  EagerCollection,
+  Values,
+  Resource,
+  SkipService,
+} from "@skipruntime/core";
+
+import { PostgresExternalService } from "@skip-adapter/postgres";
+
+type Post = {
+  author_id: number;
+  title: string;
+  url: string;
+  body: string;
+  date: number;
+};
+
+type User = {
+  name: string;
+  email: string;
+};
+const unknownUser: User = { name: "unknown author", email: "unknown email" };
+
+type Upvote = {
+  post_id: number;
+  user_id: number;
+};
+
+type PostWithUpvoteIds = Post & { upvotes: number[]; author: User };
+
+type Session = User & {
+  user_id: number;
+};
```

Comment:
Maybe just me being paranoid, but I would move these type definitions out into a separate file, and use that file also for the follower, to make sure that the type definitions stay in sync between them.

Reply:
Hmm, easier said than done with docker/typescript -- can't just import or symlink it from outside their respective root dirs without breaking things.  Maybe that's a data point in favor of "single service definition, multiple runners" but in any case I'm going to set it aside for now :)